File: backend/app/domain/entities/email.py
"""Email and Attachment domain entities. Zero framework dependency."""

from dataclasses import dataclass, field
from datetime import datetime
from uuid import UUID


@dataclass
class Attachment:
    id: UUID
    email_id: UUID
    gmail_attachment_id: str
    filename: str
    mime_type: str
    size: int


@dataclass
class Email:
    id: UUID
    thread_id: UUID
    user_id: UUID
    gmail_message_id: str
    sender: str
    recipients: list[str]
    cc: list[str]
    bcc: list[str]
    subject: str | None
    snippet: str
    body_text: str | None
    body_html: str | None
    received_at: datetime
    is_read: bool
    is_starred: bool
    has_attachments: bool
    label_ids: list[str]
    created_at: datetime
    updated_at: datetime
    attachments: list[Attachment] = field(default_factory=list)

File: frontend/src/types/email.ts
export interface AttachmentSummary {
  id: string;
  filename: string;
  mime_type: string;
  size: number;
}

export interface EmailSummary {
  id: string;
  gmail_message_id: string;
  thread_id: string;
  sender: string;
  subject: string | null;
  snippet: string;
  received_at: string;
  is_read: boolean;
  is_starred: boolean;
  has_attachments: boolean;
}

export interface EmailDetail extends EmailSummary {
  recipients: string[];
  cc: string[];
  bcc: string[];
  body_text: string | null;
  body_html: string | null;
  label_ids: string[];
  attachments: AttachmentSummary[];
}

export interface AIUnderstandingResult {
  category: string;
  intent: string;
  urgency: string;
  sentiment: string;
  entities: Array<{ type: string; value: string }>;
  summary: string;
  confidence: number;
}